include/pagmo/algorithms/gwo.hpp

```diff
@@ -0,0 +1,351 @@
+            {   
+              tmp.push_back(drng(m_seed)*(ub-lb)+lb);
+            }
+                 agents_position.push_back(tmp);
+            }
+
+        // Main gwo iterations
+        for (decltype(m_gen) gen = 1u; gen <= m_gen; ++gen) 
+        {
+             // Start of the loop through each agent
+            for (decltype(NP) i = 0u; i < NP; ++i){
+                  // clip position value that goes beyond search space          
+              for(decltype(dim) k = 0; k < dim; ++k)
+                boost::algorithm::clamp(agents_position[i][k], lb, ub);
+
+
+                auto newfitness = prob.fitness(agents_position[i]);
+
+                // Update alpha, beta and delta
+                if (newfitness < alpha_score)
+                    {
+                        alpha_score = newfitness;
+                        alpha_pos = agents_position[i];
+                    }
+
+                if(newfitness > alpha_score && newfitness < beta_score){
+                    beta_score = newfitness;
+                    beta_pos = agents_position[i];
+                }
+
+                if(newfitness > alpha_score && newfitness > beta_score && newfitness <  delta_score){
+                    delta_score = newfitness;
+                    delta_pos = agents_position[i];
+
+                }
+               
+        
+            }
+             a= 2-gen*((2)/m_gen);
+
+           
+            for (decltype(m_agent) i = 0u; i < m_agent; ++i){
+                  // Encircling prey and attack
+              for(decltype(dim) k = 0; k < dim; ++k){
+                    r1 = drng(m_seed);
+                    r2 = drng(m_seed);
+
+                    a_vector[0] = 2*a*r1-a;  // Equation (3.3)
+                    c_vector[0] = 2*r2; // Equation (3.4)
+
+                    d_vector[0] = std::cmath::abs(c_vector[0] * alpha_pos[k]- agents_position[i][k]); // Equation (3.5)-part 1
+                    x_vector[0] = alpha_pos[k]- a_vector[0] * d_vector[0]; // Equation (3.6)-part 1
+
+                    r1 = drng(m_seed);
+                    r2 = drng(m_seed);
+
+                    a_vector[1] =2*a*r1-a; 
+                    c_vector[1]=2*r2;
+
+                    d_vector[1]=abs(c_vector[1]*beta_pos[k]-agents_position[i][k]); // Equation (3.5)-part 2
+                    x_vector[1]= beta_pos[k]-a_vector[1] * d_vector[1]; // Equation (3.6)-part 2       
+                
+                    r1=random.random()
+                    r2=random.random() 
+                
+                
+                    a_vector[2] =2*a*r1-a; 
+                    c_vector[2]=2*r2;
+
+                    d_vector[2]=abs(c_vector[2]*delta_pos[k]-agents_position[i][k]); // Equation (3.5)-part 3
+                    x_vector[2]= delta_pos[k]-a_vector[2] * d_vector[2]; // Equation (3.6)-part 3          
+                
+                agents_position[i][k] = (x_vector[0]+x_vector[1]+x_vector[2])/3;  // Equation (3.7)
+
```

Comment:
here is where I would compute the fitness (as the positions have changed) and thus update alpha beta gamma and call the ```pop.set_xf``` method that would also update the pop without causing further fitness evaluations.